### PR TITLE
Allow multiclass f-score

### PR DESCRIPTION
**Pull request by JoshChristie**
Hi!

I made an attempt to allow the f-score to support multi-classifications.

I'm thinking about implementing the same for cohens kappa score. Is that okay with you? Is refactor-0.6 the right branch to be working on?

### CODE REVIEW

File: src/classification/multiclass.jl [hunk 1]
```diff
@@ -449,19 +449,68 @@ function f_score(targets::AbstractVector,
...
 """
-    f1_score(target, output, [encoding]) -> Float64
+    f1_score(targets, outputs, [encoding], [avgmode])
 
 Same as [`f_score`](@ref), but with `β` fixed to 1.
 """
-f1_score(target, output) = f_score(target, output, 1.0)
-f1_score(target, output, enc) = f_score(target, output, enc, 1.0)
+f1_score(targets, outputs) = f_score(targets, outputs, 1.0)
+f1_score(targets, outputs, enc) = f_score(targets, outputs, enc, 1.0)
```
Comment (Evizero) [inline src/classification/multiclass.jl]: probably safer to do `enc::LabelEncoding` here

File: test/classification.jl [hunk 2]
```diff
@@ -75,6 +75,17 @@ for (fun, mask) = ((true_positives,  (0,0,0,1)),
     end
 end
 
+# f_score
+
+y_true = [0, 0, 0, 1, 1, 1, 1, 1, 2, 2, 2]
+y_pred = [0, 1, 0, 1, 0, 1, 0, 1, 2, 2, 1]
+@test f_score(y_true, y_pred) == Dict{Int, Float64}(0 => 0.5714285714285715, 1 => 0.6, 2=>0.8)
```
Comment (Evizero) [inline test/classification.jl:82]: could you box the `f_score(...)` into `@inferred(f_score(...))` to test that they are type stable?

**Review by Evizero — CHANGES_REQUESTED** [review]
This looks awesome. Thanks for doing this!

Just two comments

File: test/classification.jl (hunk 2 above)
Comment (JoshChristie) [reply test/classification.jl:82]: There seems to be a code stability issue in `LabelEnc.NativeLabels`. I wasn't able to workout a solution to fix it.

Here is an example of the issue.
```
julia> using MLLabelUtils
julia> @code_warntype LabelEnc.NativeLabels([0,1,2])
Variables:
  #self#::Any
  label::Array{Int64,1}

Body:
  begin 
      return (MLLabelUtils.LabelEnc.NativeLabels)(label::Array{Int64,1}, (Core.apply_type)(MLLabelUtils.LabelEnc.Val, (Base.arraylen)(label::Array{Int64,1})::Int64)::Type{Val{_}} where _)::MLLabelUtils.LabelEnc.NativeLabels{Int64,_} where _
  end::MLLabelUtils.LabelEnc.NativeLabels{Int64,_} where _
```

Comment (Evizero) [reply test/classification.jl:82]: Right. That's expected. Does it poison return type inference? If so we could fix that by annotating the return value for `f_score`. 

But this isn't high priority. I could do that some other time. Maybe just box those lines into `@inferred`, where `@inferred(f_score(...))` works right now.

**Comment by Evizero** [comment]
Thanks. sorry for the delay. I lost track

**Pull request merged**